orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -6599,3 +6602,86 @@ def run_step(model, attn_weight):
     assert to_attr.name == "to"
     to_value = to_attr.i
     assert to_value == pytorch_type_to_onnx_dtype(softmax_compute_type), "Cast to attribute is not as expected"
+
+
+@pytest.mark.parametrize("memory_optimization_level", [None, 0, 1, 2])
+@pytest.mark.parametrize("allow_gradient_checkpoint_export", [None, 0, 1])
+@pytest.mark.parametrize("fx", ["torch", "deepspeed"])
+def test_auto_enable_layerwise_recompute(memory_optimization_level, allow_gradient_checkpoint_export, fx, caplog):
+    if fx == "deepspeed":
+        try:
+            import deepspeed
+
+            checkpoint = deepspeed.checkpointing.checkpoint
+        except ImportError:
+            # skip if deepspeed is not installed (in amd CI)
+            pass
+    elif fx == "torch":
+        from torch.utils.checkpoint import checkpoint
+
+    original_val = os.environ.get("ORTMODULE_MEMORY_OPT_LEVEL", None)
+    original_val_allow_gradient_checkpoint_export = os.environ.get("ORTMODULE_ALLOW_AUTOGRAD_CHECKPOINT", None)
+
+    if memory_optimization_level is not None:
```

Comment:
I think we need to list the supported combinations of these flags and their meanings.
E.g., 
memory_optimization_level=1, allow_gradient_checkpoint_export=0 => does something
memory_optimization_level=2, allow_gradient_checkpoint_export=0 => does something
memory_optimization_level=2, allow_gradient_checkpoint_export=1 => does something

etc..

Reply:
Added.